documentation/specs/proposed/BuildCheck.md

```diff
@@ -0,0 +1,405 @@
+
+Out of scope for configuration:
+* opt-out of analysis on code-level (analogy to C# pragmas, but within msbuild xml files).
+* lower granularity of `.editorconfig` settings other than whole projects.
+* attributing configuration to a .sln file and expecting it will apply to all contained projects.
+* Support for multiple [custom configurations](#custom-configuration-declaration) within a single build for a single rule. (Not to be mixed with [standardized configuration](#standardized-configuration-declaration) - which can be configured freely per project) If a custom configuration will be used, it will need to be specified identically in each explicit configurations of the rule. This is chosen so that there are no implicit requirements on lifetime of the analyzer or analyzer instancing – each analyzer will be instantiated only once per build (this is however something that will very likely change in future versions – so authors are advised not to take hard dependency on single instance policy).
+
+### User Configurable Options
+
+Initial version of BuildCheck plans a limited set of options configurable by user (via `.editorconfig`) by which users can override default configuration of individual analyzer rules.
+
+#### Enablement
+
+Boolean option `IsEnabled` will be available to allow users to disable/enable particualr rule.
+
+Different rules of a single analyzer can have different enabledment status configured.
+
+If all the rules from a single analyzer are disabled - analyzer won't be given any data for such configured part of the build (specific project or a whoe build). If analyzer have some rules enabled and some disabled - it will be still fed with data, but the reports will be post-filtered.
+
+Same rule can have different enablement status for different projects.
+
+#### Severity
+
+Option `Severity` with following values will be available:
+
+* `Message`
+* `Warning`
+* `Error`
+
+Configuration will dictate transformation of the analyzer report to particular build output type (message, warning or error).
+
+Different rules of a single analyzer can have different severities configured. Same rule can have different severities for different projects.
+
+#### Scope of Analysis
+
+Option `EvaluationAnalysisScope` with following possible options will be avaialable:
+* `AnalyzedProjectOnly` - Only the data from currently analyzed project will be sent to the analyzer. Imports will be discarded.
+* `AnalyzedProjectWithImportsFromCurrentWorkTree` - Only the data from currently analyzed project and imports from files under the entry project or solution will be sent to the analyzer. Other imports will be discarded.
+* `AnalyzedProjectWithImportsWithoutSdks` - Imports from SDKs will not be sent to the analyzer. Other imports will be sent.
```

Comment:
So this is "include NuGet-delivered logic but not SDK-delivered logic"? What's the use case for that?

Reply:
I'm still unsure if we need this.

But I was thinking about a scenario of local development loop of package - where devs still might need to go the nuget roundtrip. But there might be a better answers to that possibly (e.g. additional development root location?)